packages/bundle-source/cache.js

```diff
@@ -238,15 +241,26 @@ export const makeBundleCache = (wr, cwd, readPowers, opts) => {
       .stat();
     assert.equal(actualBundleTime.toISOString(), bundleTime);
     const moduleRd = wr.readOnly().neighbor(moduleSource);
-    const actualTimes = await Promise.all(
-      contents.map(async ({ relativePath }) => {
-        const itemRd = moduleRd.neighbor(relativePath);
-        /** @type {import('fs').Stats} */
-        const { mtime } = await itemRd.stat();
-        return { relativePath, mtime: mtime.toISOString() };
-      }),
+    const actualStats = await Promise.all(
+      contents.map(
+        async ({ relativePath, mtime: priorMtime, size: priorSize }) => {
+          const itemRd = moduleRd.neighbor(relativePath);
+          /** @type {import('fs').Stats} */
+          const { mtime, size } = await itemRd.stat();
```

Comment:
Does this consume an unbounded number of file descriptors? Worth fixing?

Reply:
Thankfully, `stat` syscall doesn’t allocate a descriptor. `writeFile` certainly does. I can follow up with a mutex like we did with Compartment Mapper’s read powers.

Reply:
Added a commit to synchronize fs.